task_manager/task_manager.cc

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "task_manager.hh"
+
+logging::logger tmlogger("task manager");
+
+future<> task_manager::stop() noexcept {
+    if (!_modules.empty()) {
+        on_internal_error(tmlogger, "Tried to stop task manager while some modules were not unregistered");
+    }
+    return make_ready_future<>();
+}
+
+future<task_manager::foreign_task_ptr> task_manager::lookup_task_on_all_shards(sharded<task_manager>& tm, task_id tid) {
+    task_manager::foreign_task_ptr res;
+    co_await parallel_for_each(boost::irange(0u, smp::count), [&tm, tid, &res] (unsigned shard) -> future<> {
+        auto local_res = co_await tm.invoke_on(shard, [tid] (const task_manager& local_tm) {
+            const auto& all_tasks = local_tm.get_all_tasks();
+            if (auto it = all_tasks.find(tid); it != all_tasks.end()) {
+                return make_foreign<task_manager::task_ptr>(it->second);
+            }
+            return make_foreign<task_manager::task_ptr>(nullptr);
+        });
+        if (!res) {
+            res = std::move(local_res);
+        } else if (local_res) {
+            on_internal_error(tmlogger, format("task_id {} found on more than one shard", tid.to_sstring()));
```

Comment:
can't we format `tid` as is?
If not, please open an issue.  we should fix it.

Reply:
This should have done it: https://github.com/scylladb/scylladb/blob/055340ae39ba7701e798ca370d01d343c28ac231/utils/UUID.hh#L284